src/main/scala/org/constellation/checkpoint/CheckpointAcceptanceService.scala

```diff
@@ -333,29 +345,48 @@ class CheckpointAcceptanceService[F[_]: Concurrent: Timer](
   def acceptObservations(cb: CheckpointBlock, cpc: Option[CheckpointCache] = None): F[Unit] =
     cb.observations.toList.traverse(o => observationService.accept(o, cpc)).void
 
-  def acceptTransactions(cb: CheckpointBlock, cpc: Option[CheckpointCache] = None): F[Unit] = {
+  def acceptTransactions(
+    cb: CheckpointBlock,
+    cpc: Option[CheckpointCache] = None,
+    txsHashToFilter: Seq[String] = Seq.empty
+  ): F[Unit] = {
     def toCacheData(tx: Transaction) = TransactionCacheData(
       tx,
       Map(cb.baseHash -> true),
       cbBaseHash = Some(cb.baseHash)
     )
 
-    val insertTX =
-      LiftIO[F].liftIO {
-        cb.transactions.toList
-          .map(tx ⇒ (tx, toCacheData(tx)))
-          .traverse {
-            case (tx, txMetadata) =>
-              dao.transactionService.accept(txMetadata, cpc) >> transferIfNotDummy(tx)
-          }
-          .void
+    cb.transactions.toList
+      .filterNot(h => txsHashToFilter.contains(h.baseHash))
+      .map(tx => (tx, toCacheData(tx)))
+      .traverse {
+        case (tx, txMetadata) => LiftIO[F].liftIO(dao.transactionService.accept(txMetadata, cpc)) >> transfer(tx)
       }
-
-    insertTX
+      .void
   }
 
-  private def transferIfNotDummy(transaction: Transaction): IO[Unit] =
-    if (!transaction.isDummy) dao.addressService.transfer(transaction).void else IO.unit
+  private def transfer(tx: Transaction): F[Unit] =
+    shouldBlockTransaction(tx).ifM(
+      logger.info(s"Transaction with hash blocked = ${tx.baseHash} : is dummy = ${tx.isDummy}"),
+      LiftIO[F].liftIO(dao.addressService.transfer(tx).void)
+    )
+
+  private def shouldBlockTransaction(tx: Transaction): F[Boolean] =
+    for {
+      isBlacklisted <- isBlacklistedAddress(tx)
+      _ <- if (isBlacklisted)
+        logger.info(s"Transaction blocked with hash = ${tx.baseHash} : from address = ${tx.src.address}")
+      else Sync[F].unit
+      _ <- if (isBlacklisted) dao.metrics.incrementMetricAsync[F]("blockedBlacklistedTxs") else Sync[F].unit
+
+      isDummy <- isDummyTransaction(tx)
+    } yield isBlacklisted || isDummy
```

Comment:
Dummy tx should be blocked?

Reply:
Yeap, We don't transfer if it is a dummy transaction
I implemented it a long time ago

Reply:
So maybe the name is wrong? shouldTransfer instead of shouldBlock?